### ISSUE TITLE
Ensure that Popup type provides dialog semantics.

### ISSUE THREAD

**Issue by yzen** [mozilla-mobile/mozilla-vpn-client#322]
This issue is related to #136 and #138 and accessibility around `Popup` QML types used across the VPN app.

`Popup` is currently used in:

* `VPNConnectionInfo`
* `VPNRemoveDevicePopup`

We need to ensure that they are semantically a Dialog with the corresponding name and accessibility events firing on show/hide (open/close).

Looks like having `Accessible.role: Accessible.Dialog` and `Accessible.name` does not yield the expected result. In fact it throws a warning  that `Accessible....` properties do not apply on something that does not inherit from `Item`, as is the case with `Popup` that inherits from `QObject` -> `QtObject`.

The content of the popup seems to land directly as children of the top level Window of the VPN app. It's not contained within the `Popup` container. This is why we implemented workarounds for this issue with setting `visible` explicitly to `false` on all elements that are not part of the `Popup` when the `Popup` is open.

Probably because of the missing dialog semantics, we still do not get the appropriate accessibility events.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/VPN-86)
┆author: Data Sync

**Comment by jess-cook03** [comment]
@lesleyjanenorton @bakulf has this been fixed? Can we close?
